dcs_deploy.py

```diff
@@ -446,31 +703,131 @@ def load_selected_config(self):
+            if self.args.ab_partition == True:
+                # setup multiple app partitions
+                self.ext_partition_layout = os.path.relpath('tools/kernel_flash/flash_l4t_nvme_rootfs_ab.xml')
+            else:
+                # setup no multiple app partitions
+                self.ext_partition_layout = os.path.relpath('tools/kernel_flash/flash_l4t_external_custom.xml')
+        else:
+            print("Unknown storage [%s]! exitting" % self.config['storage'])
+            exit(9)
+        # fix default rootdev to external  (or internal) for orin. There is NFS used to flash
+        if self.config['device'] == 'orin_nx' or self.config['device'] == 'orin_nx_8gb' or self.config['device'] == 'orin_nano_8gb' or self.config['device'] == 'orin_nano_4gb':
+            self.rootdev = "external" #specify "internal" - boot from  on-board device (eMMC/SDCARD), "external" - boot from external device. For more see flash.sh examples
+
+    def generate_images(self):
+        self.prepare_status.change_group("images")
+        # check commandline parameter if they are same as previous and images are already generated skip generation
+        if self.prepare_status.is_identifier_same_as_prev(["--regen", "--force"]) and self.prepare_status.get_status() == True:
+            print("Images already generated! Skipping generating images!")
+            return 0
+
+        self.prepare_status.set_processing_step("generate_images")
+        print("-"*80)
+        print("Generating images! ...")
+        ret = -2
+
+        # Note: --no-flash parameter allows us to only generate images which will be used for flashing new devices
+        # flash internal emmc"
+        if self.config['storage'] == 'emmc':
+            ret = cmd_exec(f"sudo ./{self.flash_script_path} --no-flash --showlogs {self.board_name} {self.rootdev}")
+        # flash external nvme drive
+        elif self.config['storage'] == 'nvme':
+            #file to check: initrdflashparam.txt - contains last enterred parameters
+            env_vars = ""
+            opt_app_size_arg = ""
+            external_only = "--external-only" # flash only external device
+            
+            if self.args.ab_partition == True:
+                env_vars = "ROOTFS_AB=1"
+                if self.args.rootfs_type == "minimal":
+                    opt_app_size = 4
+                else:
+                    opt_app_size = 8
+                opt_app_size_arg = f"-S {opt_app_size}GiB"
+                external_only = "" # flash internal and external device
+                #self.rootdev = "external" # set UUID device in kernel commandline: rootfs=PARTUUID=<external-uuid>
+
+            if self.args.app_size is not None:
+                opt_app_size_arg = f"-S {self.args.app_size}GiB"
+
+            if self.config['device'] == 'orin_nx' or self.config['device'] == 'orin_nx_8gb' or self.config['device'] == 'orin_nano_8gb' or self.config['device'] == 'orin_nano_4gb':
```

Comment:
More compact and more readable version nr. 2
```suggestion
            if self.config['device'] in ['orin_nx', 'orin_nx_8gb', 'orin_nano_8gb', 'orin_nano_4gb']:
```